### DISCUSSION TITLE
Create table from struct

### DISCUSSION THREAD

**Discussion by barsdeveloper** [diesel-rs/diesel#4285] [Q&A]
Is there in Diesel something similar to  `schema.create_table_from_entity(entity_object)` from Sea ORM that  gives a query that when executed creates a table suitable for a Entity. With Sea ORM I was using a code first approach that on the first run sets up the database to be coherent with the entities I have. Is there something similar in Diesel that works programmatically (without using a CLI).

**Comment by weiznich** [comment]
We do currently not provide such a functionality, but we are open to accept contributions that add something like that based on the `table!` definitions in the `schema.rs` file. If you are interested to work on this feature I can provide more details on where to start and how to implement this.

**Reply by barsdeveloper** [reply]
Great, thanks. I will try and contact you.
